omen2/table.py

```diff
@@ -122,6 +126,10 @@ def _add(self, obj: U, upsert: bool) -> U:
     def _notx_add(self, obj: U, upsert: bool = False) -> U:
         obj._bind(table=self)
         obj._commit(upsert=upsert)
+        if upsert:
+            with suppress(OmenNoPkError):
```

Comment:
could you add a comment on when this is expected

Reply:
i will write a test case.  it's really an edge case.   i think i should catch and clear the cache.

Reply:
yeah, it's impossible.  adding support for "no pk" situations is a bigger effort and not terribly useful